database/models.py

```diff
@@ -0,0 +1,179 @@
+        name (CharField): Full name of user
+        email (CharField): Email
+        address (CharField): Full street address (For where checks can be sent to user)
+        cert (CharField): Location of the BOSO purchase certification on the local filesystem. It
+            can be false, if it was deleted on not uploaded during registration.
+        revoke_counter (IntegerField): A counter that increments whenever tokens are revoked, thus
+            invalidating all old JWT.
+    """
+
+    username = CharField(max_length=16, primary_key=True)
+    password = CharField(max_length=512)
+    name = CharField(max_length=200)
+    email = CharField(max_length=100)
+    address = CharField(max_length=400)
+    cert = CharField(max_length=300, null=True)
+    revoke_counter = IntegerField(default=0)
+
+class Organizations(Base):
+    """ The Organizations table. Every budget, purchase and income must be
+    associated with an organization. An organization needs to have a name, and
+    can optionally have a parent organization.
+
+    Attributes:
+        name (CharField): Organization name
+        parent (ForeignKey on name): The parent of this organization (optional)
+    """
+
+    name = CharField(max_length=100, primary_key=True)
+    parent = ForeignKeyField('self', null=True, default=None, db_column='parent')
+
+class Budgets(Base):
+    """ A budget for an organization. An organization can have multiple
+    budgets, but they are unique on the organization, name and year columns.
+
+    Attributes:
+        organization (ForeignKey on an Organizations): Who owns the organization
+        name (CharField): Name of this budget
+        year (IntegerField): The fiscal year this is valid for (Starts at Fall Semester)
+        amount (DecimalField): The monetary amount for this budget
+    """
+
+    organization = ForeignKeyField(Organizations, db_column='organization')
+    name = CharField(max_length=250)
+    year = IntegerField()
+    amount = DecimalField(decimal_places=2)
+
+    class Meta:
+        # Create a unique index on organization, name and year
+        indexes = (
+            (('organization', 'name', 'year'), True),
+        )
+
+class Purchases(Base):
+    """ The Purchase table. Holds information about a purchase.
+
+    Attributes:
+        username (ForeignKey on a User): Who made this purchase
+        approved_by (ForeignKey on a User): Who approved this purchase
+        budget (ForeignKey on a budget): What budget this belongs to
+        purchase_date (DateTimeField): When it was purchased
+        item (CharField): What was purchased
+        reason (CharField): Why was it purchased
+        vendor (CharField): Who was it purchased from
+        amount (DecimalField): Monetary amount of purchase
+        status (IntegerField): Reimbursement status: (
+            1: 'Requested', 2: 'Approved', 3: 'Denied', 4: 'Purchased',
+            5: 'Processing Reimbursement', 6: 'Reimbursed'
```

Comment:
MySQL enums :P 

I could look into declaring an enum type like I did with the modified time. Little trickier since you need to somehow declare the possible values